src/amo/components/Header/index.js

```diff
@@ -78,11 +121,22 @@ export class HeaderBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state) {
-  return { isHomePage: state.viewContext.context === VIEW_CONTEXT_HOME };
-}
+export const mapStateToProps = (state) => {
+  return {
+    api: state.api,
+    isHomePage: state.viewContext.context === VIEW_CONTEXT_HOME,
+    username: state.user.username,
+    isAuthenticated: isUserAuthenticated(state),
+  };
+};
+
+export const mapDispatchToProps = (dispatch) => ({
+  handleLogOut: ({ api }) => {
+    return logOutFromServer({ api }).then(() => dispatch(logOutUser()));
+  },
+});
```

Comment:
I am a bit sad to duplicate the logic of the `AuthenticateButton`. Is that OK? Should I handle things differently?